src/Token.php

```diff
@@ -59,15 +59,22 @@ public function getEndPosition() {
         return $this->fullStart + $this->length;
     }
 
-    public static function getTokenKindNameFromValue($kindName) {
-        $constants = (new \ReflectionClass("Microsoft\\PhpParser\\TokenKind"))->getConstants();
-        foreach ($constants as $name => $val) {
-            if ($val == $kindName) {
-                $kindName = $name;
-                break;
-            }
+    /**
+     * @return string[] - [int $kind_valueMaps int (Kind value) to string (Kind name)
+     */
+    private static function getTokenKindNameFromValueMap() {
+        static $mapToKind;
+        if ($mapToKind === null) {
+            $constants = (new \ReflectionClass("Microsoft\\PhpParser\\TokenKind"))->getConstants();
+            $mapToKind = \array_flip($constants);
         }
-        return $kindName;
+        return $mapToKind;
+    }
+
+    /** @return int (Or string, if $kindName wasn't found) */
+    public static function getTokenKindNameFromValue($kindName) {
+        $mapToKind = self::getTokenKindNameFromValueMap();
+        return $mapToKind[$kindName] ?? $kindName;
```

Comment:
The variable names are confusing here (and they were like that before). $kindName the parameter is really the value? And I think it should return "" or null if it isn't found.

Reply:
Was in a rush in the morning, noticed that $kindName was wrong but didn't fix it (Threw the rest of the documentation it was based off of off).

This appears to be used in generating debug data from tests, and I don't plan on changing the function behavior in this PR.

Reply:
Sure. Thanks!